### PR TITLE
Added react rick and Morty app

### PR DESCRIPTION
**Pull request by olek557**
### React Friends App

[Demo](https://olek557.github.io/react-rick-and-morty-gallery/index.html) |
[Code base](https://github.com/olek557/react-rick-and-morty-gallery)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/olek557/react-friends-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      result = list;
+    }
+    else if (option === 'location') {
+      result = list.filter((listItem) => {
+        return listItem.location.name.toLowerCase() == value.toLowerCase();
+      });
+    }
+    else {
+      result = list.filter((listItem) => {
+        return listItem[option].toLowerCase() === value;
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/App.js]: `value.toLowerCase()`? And if so - you can combine this and `location` into one function. DRY

File: submissions/olek557/react-friends-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      });
+    }
+    return result;
+  }
+
+  generateFilteredList = (newFilteringOptions) => {
+    let listToRender = this.state.originalList;
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/App.js]: `{ originalList }`

File: submissions/olek557/react-friends-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    const { nameSearch, genderFilter, speciesFilter, locationFilter } = newFilteringOptions;
+    if (nameSearch) {
+      listToRender = this.searchInList(listToRender, 'name', nameSearch);
+    }
+    if (genderFilter) {
+      listToRender = this.filterlist(listToRender, 'gender', genderFilter);
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/App.js]: Try to find out logic, with which you can remove this string flags.

File: submissions/olek557/react-friends-app/src/App.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    if (locationFilter) {
+      listToRender = this.filterlist(listToRender, 'location', locationFilter);
+    }
+    return listToRender;
+  }
+
+
+  performFiltering = (event) => {
+    const input = event.target;
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/App.js]: `{ target }`

File: submissions/olek557/react-friends-app/src/App.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    const newFilteringOptions = Object.assign({}, this.state.filteringOptions);
+    newFilteringOptions[input.name] = input.value;
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/App.js]: Can be reduced to one operation with spread operator. You should know features of the language

File: submissions/olek557/react-friends-app/src/Filters.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import './Filters.css';
+
+export class Filters extends Component {
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/Filters.js]: Can be converted to functional component

File: submissions/olek557/react-friends-app/src/api.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const url = "https://rickandmortyapi.com/api/character/";
+
+
+let getData = async () => {
+  const response = await fetch(url);
```
Comment (zonzujiro) [inline submissions/olek557/react-friends-app/src/api.js]: And what if `fetch` fail?

**Comment by zonzujiro** [comment]
Ok, looks like dead. I will close it for now. Fill free to open new one

File: submissions/olek557/react-friends-app/src/App.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    let result;
+    if (value === 'all') {
+      result = list;
+    }
+    else if (option === 'location') {
+      result = list.filter((listItem) => {
+        return listItem.location.name.toLowerCase() == value.toLowerCase();
+      });
+    }
+    else {
```
[the remark this answers is not in the document]
Comment (olek557) [reply submissions/olek557/react-friends-app/src/App.js]: Thanks. It was my fault

File: submissions/olek557/react-friends-app/src/Filters.js (hunk 6 above)
Comment (olek557) [reply submissions/olek557/react-friends-app/src/Filters.js]: Make sense. Thanks

File: submissions/olek557/react-friends-app/src/api.js (hunk 7 above)
Comment (olek557) [reply submissions/olek557/react-friends-app/src/api.js]: please check this.
Thanks

**Pull request closed without merging**